src/settings/handlers/ConfigSettingsHandler.ts

```diff
@@ -24,9 +24,24 @@ import {isNullOrUndefined} from "matrix-js-sdk/src/utils";
  * roomId parameter.
  */
 export default class ConfigSettingsHandler extends SettingsHandler {
+    public constructor(private featureNames: string[]) {
+        super();
+    }
+
     public getValue(settingName: string, roomId: string): any {
         const config = SdkConfig.get() || {};
 
+        if (this.featureNames.includes(settingName)) {
+            const labsConfig = config["features"] || {};
+            const val = labsConfig[settingName];
+            if (isNullOrUndefined(val)) return null; // no definition at this level
+            if (val === true || val === false) return val; // new style: mapped as a boolean
+            if (val === "enable") return true; // backwards compat
+            if (val === "disable") return false; // backwards compat
+            if (val === "labs") return null; // backwards compat, no override
```

Comment:
Should this log a warning that you probably want `showLabsSetting` now?

Reply:
normally yes, but tbh I have zero confidence that deprecation warnings in the console help us. The deprecation warning about using "weird" homeserver config options for instance is ignored by everyone, including us.

I'd almost rather keep it as an undocumented feature than deprecate it.

Reply:
another rationale is that this is called *very* often, so would probably be more annoying than helpful :(

Reply:
Right, seems fine as-is then.